pkg/csi/service/wcpguest/controller.go

```diff
@@ -280,19 +338,48 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 	if err != nil {
 		msg := fmt.Sprintf("Validation for Delete Volume Request: %+v has failed. Error: %+v", *req, err)
 		log.Error(msg)
+		volumeControlOpsHistVec.WithLabelValues(common.PrometheusUnknownVolumeType, common.PrometheusDeleteOpType,
+			common.PrometheusFailStatus).Observe(time.Since(start).Seconds())
 		return nil, err
 	}
+	// Retrieve Supervisor PVC
+	svPVC, err := c.supervisorClient.CoreV1().PersistentVolumeClaims(c.supervisorNamespace).Get(ctx, req.VolumeId, metav1.GetOptions{})
```

Comment:
Have we validated whether `Get()` call adds up few `ms` and also affect performance benchmarks for `DeleteVolume()` calls ?

Reply:
Yes, the get here should hardly add any time in the overall workflow. I see that the overall delete takes few milliseconds to complete on average with this code change which seems to be fine.

The way I see it is that the advantage of exposing the metrices per volume type and the ability to monitor and narraw down the problem by volume type outweighs the small perf impact(in ms).